src/lib/customPropTypes.tsx

```diff
@@ -28,15 +31,15 @@ export const domNode = (props, propName) => {
  * Useful for very large lists of options (e.g. Icon name, Flag name, etc.)
  * @param {string[]} suggestions An array of allowed values.
  */
-export const suggest = suggestions => {
+export const suggest = (suggestions: string[]) => {
   if (!Array.isArray(suggestions)) {
     throw new Error('Invalid argument supplied to suggest, expected an instance of array.')
   }
 
-  const findBestSuggestions = _.memoize(str => {
+  const findBestSuggestions = _.memoize((str: string) => {
     const propValueWords = str.split(' ')
 
-    return _.flow(
+    return _.flow<string[], SuggestProps[], SuggestProps[], SuggestProps[]>(
```

Comment:
the same type correctness effect could be achieved a bit easier - it is possible to omit type specs for `flow<...>`, and introduce type for the `map` function only:

```ts
...
 _.map((suggestion: string) => {
...
```

The benefit is that we won't need to introduce types to `flow` for each new function added. This would be the result code:

```ts
_.flow(
      _.map((suggestion: string) => {
        const suggestionWords = suggestion.split(' ')

        const propValueScore = _.flow(
          _.map(x => _.map(y => leven(x, y), suggestionWords)),
          _.map(_.min),
          _.sum,
        )(propValueWords)

        const suggestionScore = _.flow(
          _.map(x => _.map(y => leven(x, y), propValueWords)),
          _.map(_.min),
          _.sum,
        )(suggestionWords)

        return { suggestion, score: propValueScore + suggestionScore }
      }),
      _.sortBy<SuggestProps>(['score', 'suggestion']),
      _.take(3),
    )(suggestions)
```

Reply:
very nice, thanks! applied your suggestion